### PR TITLE
Remove concrete GazeProvider reference to allow custom providers

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview
This PR moves UpdateGazeInfoFromHit into the interface definition to remove a cast in FocusProvider to the concrete type.

It also performs some formatting / comment updates to match current MRTK-style more.

## Changes
- Fixes: #5545 (at least, as far as GazeProvider is concerned. There's also the note about potentially defining custom GazePointers, like other pointers are handled)

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs [hunk 1]
```diff
@@ -163,8 +163,6 @@ public IMixedRealityInputSource GazeInputSource
 
         private Ray latestEyeGaze = default(Ray);
         private DateTime latestEyeTrackingUpdate = DateTime.MinValue;
-        private bool? isEyeCalibrated = null;
-
         private readonly float maxEyeTrackingTimeoutInSeconds = 2.0f;
 
         #region InternalGazePointer Class
```
Comment (Kjakubzak) [inline Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs]: I think that having the InternalGazePointer class be a self-contained entity within the GazeProvider can lead to confusion for developers; especially when there's a UI element in the editor that allows them to define pointers (other than for gaze). 

It doesn't need to be addressed in this PR, but can a work item be opened up to address this concern in the future? While being able to define your own GazeProvider means you also are able to define your own InternalGazePointer; it isn't very easily discoverable.

Comment (Kjakubzak) [reply Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs]: Also, in the OnPreSceneQuery function for the internal gaze pointer; there's this line:
`gazeProvider.HitPosition = Rays[0].Origin + (gazeProvider.lastHitDistance * Rays[0].Direction);
`
Which then gets blown away by the call to UpdateGazeInfoFromHit() from the FocusProvider.

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs]: Agreed! I'll file an issue about InternalGazePointer

Comment (keveleigh) [reply Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs]: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/5612

**Pull request merged**